### PR TITLE
Updated readme and split out advanced config guide.

### PR DESCRIPTION
**Pull request by killianmuldoon**
This PR reworks the main README, moving the configuration set up and the VF set up to a separate docs folder. It also restructures the README to focus on the overall Kubernetes solution and point to the Device Plugin repo as the source of truth for orchestrating VFs in Kubernetes.

### CODE REVIEW

**Comment by killianmuldoon** [comment]
@zshi-redhat @ahalim-intel One questionable inclusion here I kept the setup doc for VFs, but I haven't updated it. Should this still be part of this repo?

**Comment by adrianchiris** [comment]
from my point its LGTM.

once https://github.com/intel/sriov-network-device-plugin/pull/240 lands , i suggest creating another PR removing `setup-vf.md` and adding a link to its sriov-network-device-plugin counterpart.

File: README.md [hunk 1]
```diff
@@ -2,243 +2,146 @@
...
-* `ipam` (dictionary, optional): IPAM configuration to be used for this network.
-* `spoofchk` (string, optional): turn packet spoof checking on or off for the VF
-* `trust` (string, optional): turn trust setting on or off for the VF
-* `link_state` (string, optional): enforce link state for the VF. Allowed values: auto, enable, disable. Note that driver support may differ for this feature. For example, `i40e` is known to work but `igb` doesn't.
-* `min_tx_rate` (int, optional): change the allowed minimum transmit bandwidth, in Mbps, for the VF. Setting this to 0 disables rate limiting. The min_tx_rate value should be <= max_tx_rate. Support of this feature depends on NICs and drivers.
+### Basic configuration parameters 
 
-* `max_tx_rate` (int, optional): change the allowed maximum transmit bandwidth, in Mbps, for the VF. 
-Setting this to 0 disables rate limiting. 
+The following parameters are generally outside the scope of SR-IOV CNI configuration, though (with the exception of ipam) they need to be included in the config.
```
Comment (zshi-redhat) [inline README.md:78]: This `outside the scope of SR-IOV CNI` is a little confusing to me, is this to say these options are generic CNI parameters that are not specific to SR-IOV CNI?

Comment (killianmuldoon) [reply README.md:78]: Yeah - I'll change this to reflect that more clearly

**Comment by killianmuldoon** [comment]
@ahalim-intel now that the doc is updated at intel/sriov-network-device-plugin#240  do you think we can merge this?

**Pull request merged**